src/bin/kuutamoctl.rs

```diff
@@ -42,27 +45,58 @@ async fn show_active_validator(kuutamo_client: &CommandClient, args: &Args) -> R
     Ok(())
 }
 
+async fn show_maintenance_status(kuutamo_client: &CommandClient) -> Result<()> {
+    Ok(println!("{}", &kuutamo_client.maintenance_status().await?))
+}
+
 /// The kuutamoctl program entry point
 #[tokio::main]
 pub async fn main() {
     let args = Args::parse();
     let kuutamo_client = CommandClient::new(&args.control_socket);
     let res = match args.action {
         Command::ActiveValidator => show_active_validator(&kuutamo_client, &args).await,
-        Command::MaintenanceShutdown(MaintenanceShutdownArgs {
+        Command::MaintenanceShutdown(MaintenanceOperationArgs {
+            minimum_length,
+            shutdown_at,
+            cancel,
+            wait,
+            ..
+        }) => {
+            if minimum_length.is_some() && shutdown_at.is_some() {
+                Err(anyhow!("We can not guarantee minimum maintenance window for a specified shutdown block height"))
+            } else {
+                let r = kuutamo_client
+                    .maintenance_operation(minimum_length, shutdown_at, cancel, false)
+                    .await;
+                if r.is_ok() && wait {
+                    while UnixStream::connect(&args.control_socket).await.is_ok() {
```

Comment:
Is this is what is happening here?

```suggestion
                    // Wait for kuutamod to terminate
                    while UnixStream::connect(&args.control_socket).await.is_ok() {
```